InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -235,11 +242,40 @@ TupleSet AstQuery::typeQuery(QList<TupleSet> input, QString type)
 
 		// TODO add the possibility to keep the input nodes:
 		auto tuple = tuples.take("ast");
-		for (const auto& t : tuple) addNodesOfType(tuples, type, t["ast"]);
+		for (const auto& t : tuple) addNodesOfType(tuples, matcher, t["ast"]);
 	}
 	return tuples;
 }
 
+TupleSet AstQuery::nameQuery(QList<TupleSet> input, QString name)
+{
+	TupleSet tuples;
+
+	QList<QPair<QString, Model::Node*>> matchingNodes;
+
+	if (scope_ == Scope::Local)
+		matchingNodes = Model::NameResolver::mostLikelyMatches(name, -1, target_);
+	else if (scope_ == Scope::Global)
+		matchingNodes = Model::NameResolver::mostLikelyMatches(name, -1);
+	else if (scope_ == Scope::Input)
+	{
+		Q_ASSERT(input.size());
+		tuples = input.takeFirst();
+
+		// TODO add the possibility to keep the input nodes:
+		auto tuple = tuples.take("ast");
+		for (const auto& t : tuple) matchingNodes << Model::NameResolver::mostLikelyMatches(name, -1, t["ast"]);
+	}
+	// If we have a type argument filter the results:
+	const QString type = argParser_.value(NODETYPE_ARGUMENT_NAMES[0]);
+	Model::SymbolMatcher matcher = matcherFor(type);
+	for (auto result : matchingNodes)
```

Comment:
Don't use the name `result` for something that is not the result of the method. It can be confusing.